### PR TITLE
Adding the scary replace method + some other prototype removals

### PR DESCRIPTION
**Pull request by thoov**
- [ ] are the error checks needed?
- [ ] tests/mixins/array_test.js is lines 20-50 needed?
- [ ] sortable-test.js?
- [ ] arranged_content_test.js line 206 replace does not work? general question about arrangedContent
- [ ] removeObjects: need help with propertyChange events
- [ ] some test are like `[].popObject() => [] + returns undefined + NO notify` should I be changing these?
- [ ]  we should be removing the catches at the beginnings of the methods

### CODE REVIEW

File: packages/ember-runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -63,14 +71,223 @@ export function objectAt(content, idx) {
...
+function arrayContentWillChange(content, startIdx, removeAmt, addAmt) {
+  var removing, lim;
+
+  // if no args are passed assume everything changes
+  if (startIdx === undefined) {
+    startIdx = 0;
+    removeAmt = addAmt = -1;
+  } else {
+    if (removeAmt === undefined) {
+      removeAmt = -1;
```
Comment (thoov) [inline packages/ember-runtime/lib/mixins/array.js]: These can just be removeAmt=-1, addAmt=-1

File: packages/ember-runtime/lib/mixins/array.js [hunk 2]
```diff
@@ -4,6 +4,13 @@
 */
 
 // ..........................................................
+// CONSTANTS
+//
+
+var OUT_OF_RANGE_EXCEPTION = "Index out of range";
```
Comment (thoov) [inline packages/ember-runtime/lib/mixins/array.js]: I took this from the mutable-array. Should abstract this in some fashion

**Comment by thoov** [comment]
@stefanpenner r?

**Comment by stefanpenner** [comment]
there are some issues with this, but i want to work on it. So merging it now.

**Pull request merged**